flox-bash/lib/commands/publish.sh

```diff
@@ -355,112 +375,129 @@ function floxPublish() {
 			esac
 		fi
 		while true; do
-			channelRepository=$(promptInput \
-				"Enter git URL (required)" \
-				"channel repository:" \
-				"$channelRepository")
 			if ensureGHRepoExists "$channelRepository" private "https://github.com/flox/floxpkgs-template.git"; then
 				[ -z "$channelRepository" ] || break
 			fi
 			warn "please enter a valid URL with which to 'flox subscribe'"
+			channelRepository=$(promptInput \
+				"Enter git URL (required)" \
+				"channel repository:" \
+				"$channelRepository")
 		done
 	fi
 	warn "channel repository: $channelRepository"
 
 	# Prompt for location(s) TO and FROM which we can (optionally) copy the
 	# built package store path(s). By default these will refer to the same
 	# URL, but can be overridden with --download-from.
-	if [ -z "$uploadTo" ]; then
+	if [[ -z "${uploadTo+1}" ]]; then
```

Comment:
Not sure .. that was part of #114 originally so I just trusted it, but you're right I don't think it works:
```
flox [flox/prerelease default] [brantley@clubsoda:~/src/etc-profiles]$ uploadTo="s3://flox-store-public?write-nar-listing=1&ls-compression=br"

flox [flox/prerelease default] [brantley@clubsoda:~/src/etc-profiles]$ echo "$uploadTo"
s3://flox-store-public?write-nar-listing=1&ls-compression=br

flox [flox/prerelease default] [brantley@clubsoda:~/src/etc-profiles]$ echo "${uploadTo+1}"
1

flox [flox/prerelease default] [brantley@clubsoda:~/src/etc-profiles]$ uploadTo=""

flox [flox/prerelease default] [brantley@clubsoda:~/src/etc-profiles]$ echo "${uploadTo+1}"
1
```
???

I'll drop the "+1" part because I'm sure that will work.